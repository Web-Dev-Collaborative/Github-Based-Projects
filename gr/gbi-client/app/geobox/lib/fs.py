# This file is part of the GBI project.
# Copyright (C) 2012 Omniscale GmbH & Co. KG <http://omniscale.com>
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import os
import sys
import ctypes
import subprocess

import logging
log = logging.getLogger(__name__)

def diskspace_available(path):
    if sys.platform == 'win32':
        free_bytes = ctypes.c_ulonglong(0)
        # GetDiskFreeSpaceExW(lpDirectoryName, lpFreeBytesAvailable (w/ user quota),
            # lpTotalNumberOfBytes, lpTotalNumberOfFreeBytes)
        ctypes.windll.kernel32.GetDiskFreeSpaceExW(ctypes.c_wchar_p(path), ctypes.pointer(free_bytes), None, None)
        return free_bytes.value
    else:
        stats = os.statvfs(path)
        return stats.f_bavail * stats.f_frsize


def diskspace_available_in_mb(path):
    free_bytes = diskspace_available(path)
    return free_bytes / 1024 / 1024

def open_file_explorer(path):
    if sys.platform == 'win32':
        os.startfile(path)
    elif sys.platform == 'darwin':
        subprocess.Popen(['open', path])
    else:
        try:
            subprocess.Popen(['xdg-open', path])
        except OSError:
            log.warn('unable to open file explorer')

if __name__ == '__main__':
    print diskspace_available(sys.argv[1])