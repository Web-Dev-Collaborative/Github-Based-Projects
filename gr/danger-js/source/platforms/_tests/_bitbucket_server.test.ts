import { resolve } from "path"
import { readFileSync } from "fs"

const fixtures = resolve(__dirname, "fixtures")

/** Returns JSON from the fixtured dir */
export const requestWithFixturedJSON = async (path: string): Promise<() => Promise<any>> => () =>
  Promise.resolve(JSON.parse(readFileSync(`${fixtures}/${path}`, {}).toString()))

class mockBitBucketServerAPI /*tslint:disable-line*/ {
  async getPullRequestInfo() {
    const fixtures = await requestWithFixturedJSON("bitbucket_server_pr.json")
    return await fixtures()
  }
  async getIssues() {
    const fixtures = await requestWithFixturedJSON("bitbucket_server_issues.json")
    return await fixtures()
  }
  async getPullRequestCommits() {
    const fixtures = await requestWithFixturedJSON("bitbucket_server_commits.json")
    return await fixtures()
  }
  async getPullRequestComments() {
    const fixtures = await requestWithFixturedJSON("bitbucket_server_comments.json")
    return await fixtures()
  }
  async getPullRequestChanges() {
    const fixtures = await requestWithFixturedJSON("bitbucket_server_changes.json")
    return await fixtures()
  }
  async getPullRequestActivities() {
    const fixtures = await requestWithFixturedJSON("bitbucket_server_activities.json")
    return await fixtures()
  }
}

// Mock API class
jest.mock("../bitbucket_server/BitBucketServerAPI", () => {
  return { BitBucketServerAPI: mockBitBucketServerAPI }
})

import { BitBucketServer } from "../BitBucketServer"
import { BitBucketServerAPI } from "../bitbucket_server/BitBucketServerAPI"

import { RepoMetaData } from "../../dsl/BitBucketServerDSL"

describe("getPlatformReviewDSLRepresentation", () => {
  let bbs: BitBucketServer
  let api: BitBucketServerAPI

  beforeEach(() => {
    api = new BitBucketServerAPI({} as RepoMetaData, { host: "fake://host" })
    bbs = new BitBucketServer(api)
  })

  it("should return the correct review title from getReviewInfo", async () => {
    const info = await bbs.getReviewInfo()
    expect(info.title).toEqual("Pull request title")
  })

  it("should get the commits of the pull request", async () => {
    const { commits } = await bbs.getPlatformReviewDSLRepresentation()
    expect(commits).toMatchSnapshot()
  })

  it("should get the activities", async () => {
    const { activities } = await bbs.getPlatformReviewDSLRepresentation()
    expect(activities).toMatchSnapshot()
  })

  it("should get the reviewer comments", async () => {
    const { comments } = await bbs.getPlatformReviewDSLRepresentation()
    expect(comments).toMatchSnapshot()
  })

  it("should get the pull request information", async () => {
    const { pr } = await bbs.getPlatformReviewDSLRepresentation()
    expect(pr).toMatchSnapshot()
  })

  it("should get the metadata", async () => {
    const dsl = await bbs.getPlatformReviewDSLRepresentation()
    expect(dsl.metadata).toMatchSnapshot()
  })

  it("should update status", async () => {
    api.postBuildStatus = jest.fn().mockReturnValue({ ok: true })
    await bbs.updateStatus(true, "message", undefined, "danger", undefined)

    expect(api.postBuildStatus).toHaveBeenCalledWith("d6725486c38d46a33e76f622cf24b9a388c8d13d", {
      state: "SUCCESSFUL",
      key: "danger",
      name: "danger",
      url: "http://danger.systems/js",
      description: "message",
    })
  })

  it("should update status with ci commit hash", async () => {
    api.postBuildStatus = jest.fn().mockReturnValue({ ok: true })
    await bbs.updateStatus(true, "message", undefined, "danger", "abc1234")

    expect(api.postBuildStatus).toHaveBeenCalledWith("abc1234", {
      state: "SUCCESSFUL",
      key: "danger",
      name: "danger",
      url: "http://danger.systems/js",
      description: "message",
    })
  })
})
