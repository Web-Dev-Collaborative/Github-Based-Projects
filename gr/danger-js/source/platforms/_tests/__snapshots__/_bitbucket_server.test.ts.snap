// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPlatformReviewDSLRepresentation should get the activities 1`] = `
Array [
  Object {
    "action": "RESCOPED",
    "added": Object {
      "commits": Array [
        Object {
          "author": Object {
            "active": true,
            "displayName": "DangerCI",
            "emailAddress": "foo@bar.com",
            "id": 2,
            "links": Object {
              "self": Array [
                Object {
                  "href": "http://localhost:7990/users/danger",
                },
              ],
            },
            "name": "danger",
            "slug": "danger",
            "type": "NORMAL",
          },
          "authorTimestamp": 1519442341000,
          "committer": Object {
            "active": true,
            "displayName": "DangerCI",
            "emailAddress": "foo@bar.com",
            "id": 2,
            "links": Object {
              "self": Array [
                Object {
                  "href": "http://localhost:7990/users/danger",
                },
              ],
            },
            "name": "danger",
            "slug": "danger",
            "type": "NORMAL",
          },
          "committerTimestamp": 1519442341000,
          "displayId": "d6725486c38",
          "id": "d6725486c38d46a33e76f622cf24b9a388c8d13d",
          "message": "Modify and remove files",
          "parents": Array [
            Object {
              "displayId": "c62ada76533",
              "id": "c62ada76533a2de045d4c6062988ba84df140729",
            },
          ],
        },
      ],
      "total": 1,
    },
    "createdDate": 1519442356495,
    "fromHash": "d6725486c38d46a33e76f622cf24b9a388c8d13d",
    "id": 61,
    "previousFromHash": "c62ada76533a2de045d4c6062988ba84df140729",
    "previousToHash": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
    "removed": Object {
      "commits": Array [],
      "total": 0,
    },
    "toHash": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "foo@bar.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "COMMENTED",
    "comment": Object {
      "author": Object {
        "active": true,
        "displayName": "DangerCI",
        "emailAddress": "foo@bar.com",
        "id": 2,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/danger",
            },
          ],
        },
        "name": "danger",
        "slug": "danger",
        "type": "NORMAL",
      },
      "comments": Array [],
      "createdDate": 1518939353345,
      "id": 10,
      "permittedOperations": Object {
        "deletable": true,
        "editable": false,
      },
      "properties": Object {
        "repositoryId": 1,
      },
      "tasks": Array [],
      "text": "

| 🚫 | Fails
| --- | --- |

> node\` failed.


| ⚠️ | Warnings
| --- | --- |

> Please add a changelog entry for your changes. You can find it in \`CHANGELOG.md\` 
> 
>  Please add your change and name to the master section.

> 0

> \\"use strict\\";
> 
> module.exports = {
>   runner: \\"jest-runner-eslint\\",
>   displayName: \\"lint\\",
>   testMatch: [\\"<rootDir>/**/*.js\\"],
>   testPathIgnorePatterns: [\\"node_modules/\\"]
> };



### Log


\`\`\`sh
Jest tests passed :+1:
Error:  SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (/Users/azz/code/danger-js/distribution/platforms/git/gitJSONToGitDSL.js:72:60)
    at step (/Users/azz/code/danger-js/distribution/platforms/git/gitJSONToGitDSL.js:32:23)
    at Object.next (/Users/azz/code/danger-js/distribution/platforms/git/gitJSONToGitDSL.js:13:53)
    at fulfilled (/Users/azz/code/danger-js/distribution/platforms/git/gitJSONToGitDSL.js:4:58)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

\`\`\`



|    |
|---:|
| _Generated by 🚫 [dangerJS](http://github.com/danger/danger-js/)_ |


[](http://danger-id-default;)
",
      "updatedDate": 1519449132488,
      "version": 23,
    },
    "commentAction": "ADDED",
    "createdDate": 1518939353345,
    "id": 52,
    "user": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/danger",
          },
        ],
      },
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "UPDATED",
    "addedReviewers": Array [],
    "createdDate": 1518937747368,
    "id": 43,
    "removedReviewers": Array [
      Object {
        "active": true,
        "displayName": "DangerCI",
        "emailAddress": "foo@bar.com",
        "id": 2,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/danger",
            },
          ],
        },
        "name": "danger",
        "slug": "danger",
        "type": "NORMAL",
      },
    ],
    "user": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/danger",
          },
        ],
      },
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "UPDATED",
    "addedReviewers": Array [
      Object {
        "active": true,
        "displayName": "DangerCI",
        "emailAddress": "foo@bar.com",
        "id": 2,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/danger",
            },
          ],
        },
        "name": "danger",
        "slug": "danger",
        "type": "NORMAL",
      },
    ],
    "createdDate": 1518929751675,
    "id": 42,
    "removedReviewers": Array [],
    "user": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/danger",
          },
        ],
      },
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "COMMENTED",
    "comment": Object {
      "author": Object {
        "active": true,
        "displayName": "test",
        "emailAddress": "foo@bar.com",
        "id": 1,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/test",
            },
          ],
        },
        "name": "test",
        "slug": "test",
        "type": "NORMAL",
      },
      "comments": Array [],
      "createdDate": 1518870689956,
      "id": 2,
      "permittedOperations": Object {
        "deletable": true,
        "editable": true,
      },
      "properties": Object {
        "repositoryId": 1,
      },
      "tasks": Array [],
      "text": "wow",
      "updatedDate": 1518870689956,
      "version": 0,
    },
    "commentAction": "ADDED",
    "commentAnchor": Object {
      "diffType": "EFFECTIVE",
      "fileType": "TO",
      "fromHash": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
      "line": 1,
      "lineType": "ADDED",
      "orphaned": false,
      "path": "banana",
      "toHash": "7fe85eb2a4dbaa70414155022abc7e5465f09547",
    },
    "createdDate": 1518870689956,
    "diff": Object {
      "destination": Object {
        "components": Array [
          "banana",
        ],
        "name": "banana",
        "parent": "",
        "toString": "banana",
      },
      "hunks": Array [
        Object {
          "destinationLine": 1,
          "destinationSpan": 1,
          "segments": Array [
            Object {
              "lines": Array [
                Object {
                  "commentIds": Array [
                    2,
                  ],
                  "destination": 1,
                  "line": "bar",
                  "source": 0,
                  "truncated": false,
                },
              ],
              "truncated": false,
              "type": "ADDED",
            },
          ],
          "sourceLine": 0,
          "sourceSpan": 0,
          "truncated": false,
        },
      ],
      "properties": Object {
        "current": true,
        "fromHash": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
        "toHash": "7fe85eb2a4dbaa70414155022abc7e5465f09547",
      },
      "source": null,
      "truncated": false,
    },
    "id": 5,
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "foo@bar.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "COMMENTED",
    "comment": Object {
      "author": Object {
        "active": true,
        "displayName": "test",
        "emailAddress": "foo@bar.com",
        "id": 1,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/test",
            },
          ],
        },
        "name": "test",
        "slug": "test",
        "type": "NORMAL",
      },
      "comments": Array [],
      "createdDate": 1518870515570,
      "id": 1,
      "permittedOperations": Object {
        "deletable": true,
        "editable": true,
      },
      "properties": Object {
        "repositoryId": 1,
      },
      "tasks": Array [],
      "text": "heyo",
      "updatedDate": 1518870515570,
      "version": 0,
    },
    "commentAction": "ADDED",
    "createdDate": 1518870515570,
    "id": 4,
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "foo@bar.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "OPENED",
    "createdDate": 1518863923424,
    "id": 3,
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "foo@bar.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
]
`;

exports[`getPlatformReviewDSLRepresentation should get the commits of the pull request 1`] = `
Array [
  Object {
    "author": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
    "authorTimestamp": 1519442341000,
    "committer": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
    "committerTimestamp": 1519442341000,
    "displayId": "d6725486c38",
    "id": "d6725486c38d46a33e76f622cf24b9a388c8d13d",
    "message": "Modify and remove files",
    "parents": Array [
      Object {
        "displayId": "c62ada76533",
        "id": "c62ada76533a2de045d4c6062988ba84df140729",
      },
    ],
  },
  Object {
    "author": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
    "authorTimestamp": 1518863882000,
    "committer": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
    "committerTimestamp": 1518863882000,
    "displayId": "c62ada76533",
    "id": "c62ada76533a2de045d4c6062988ba84df140729",
    "message": "add banana",
    "parents": Array [
      Object {
        "displayId": "8942a1f75e4",
        "id": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
      },
    ],
  },
]
`;

exports[`getPlatformReviewDSLRepresentation should get the metadata 1`] = `undefined`;

exports[`getPlatformReviewDSLRepresentation should get the pull request information 1`] = `
Object {
  "author": Object {
    "approved": false,
    "role": "AUTHOR",
    "status": "UNAPPROVED",
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "user@email.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
  "closed": false,
  "createdDate": 1518863923273,
  "fromRef": Object {
    "displayId": "foo",
    "id": "refs/heads/foo",
    "latestCommit": "d6725486c38d46a33e76f622cf24b9a388c8d13d",
    "repository": Object {
      "forkable": true,
      "id": 1,
      "links": Object {
        "clone": Array [
          Object {
            "href": "http://test@localhost:7990/scm/proj/repo.git",
            "name": "http",
          },
          Object {
            "href": "ssh://git@localhost:7999/proj/repo.git",
            "name": "ssh",
          },
        ],
        "self": Array [
          Object {
            "href": "http://localhost:7990/projects/PROJ/repos/repo/browse",
          },
        ],
      },
      "name": "Repo",
      "project": Object {
        "id": 1,
        "key": "PROJ",
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/projects/PROJ",
            },
          ],
        },
        "name": "Project",
        "public": false,
        "type": "NORMAL",
      },
      "public": false,
      "scmId": "git",
      "slug": "repo",
      "state": "AVAILABLE",
      "statusMessage": "Available",
    },
  },
  "id": 1,
  "links": Object {
    "self": Array [
      Object {
        "href": "http://localhost:7990/projects/PROJ/repos/repo/pull-requests/1",
      },
    ],
  },
  "locked": false,
  "open": true,
  "participants": Array [
    Object {
      "approved": false,
      "role": "PARTICIPANT",
      "status": "UNAPPROVED",
      "user": Object {
        "active": true,
        "displayName": "DangerCI",
        "emailAddress": "user@email.com",
        "id": 2,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/danger",
            },
          ],
        },
        "name": "danger",
        "slug": "danger",
        "type": "NORMAL",
      },
    },
  ],
  "reviewers": Array [],
  "state": "OPEN",
  "title": "Pull request title",
  "toRef": Object {
    "displayId": "master",
    "id": "refs/heads/master",
    "latestCommit": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
    "repository": Object {
      "forkable": true,
      "id": 1,
      "links": Object {
        "clone": Array [
          Object {
            "href": "http://test@localhost:7990/scm/proj/repo.git",
            "name": "http",
          },
          Object {
            "href": "ssh://git@localhost:7999/proj/repo.git",
            "name": "ssh",
          },
        ],
        "self": Array [
          Object {
            "href": "http://localhost:7990/projects/PROJ/repos/repo/browse",
          },
        ],
      },
      "name": "Repo",
      "project": Object {
        "id": 1,
        "key": "PROJ",
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/projects/PROJ",
            },
          ],
        },
        "name": "Project",
        "public": false,
        "type": "NORMAL",
      },
      "public": false,
      "scmId": "git",
      "slug": "repo",
      "state": "AVAILABLE",
      "statusMessage": "Available",
    },
  },
  "updatedDate": 1519442356495,
  "version": 2,
}
`;

exports[`getPlatformReviewDSLRepresentation should get the reviewer comments 1`] = `
Array [
  Object {
    "action": "RESCOPED",
    "added": Object {
      "commits": Array [
        Object {
          "author": Object {
            "active": true,
            "displayName": "DangerCI",
            "emailAddress": "foo@bar.com",
            "id": 2,
            "links": Object {
              "self": Array [
                Object {
                  "href": "http://localhost:7990/users/danger",
                },
              ],
            },
            "name": "danger",
            "slug": "danger",
            "type": "NORMAL",
          },
          "authorTimestamp": 1519442341000,
          "committer": Object {
            "active": true,
            "displayName": "DangerCI",
            "emailAddress": "foo@bar.com",
            "id": 2,
            "links": Object {
              "self": Array [
                Object {
                  "href": "http://localhost:7990/users/danger",
                },
              ],
            },
            "name": "danger",
            "slug": "danger",
            "type": "NORMAL",
          },
          "committerTimestamp": 1519442341000,
          "displayId": "d6725486c38",
          "id": "d6725486c38d46a33e76f622cf24b9a388c8d13d",
          "message": "Modify and remove files",
          "parents": Array [
            Object {
              "displayId": "c62ada76533",
              "id": "c62ada76533a2de045d4c6062988ba84df140729",
            },
          ],
        },
      ],
      "total": 1,
    },
    "createdDate": 1519442356495,
    "fromHash": "d6725486c38d46a33e76f622cf24b9a388c8d13d",
    "id": 61,
    "previousFromHash": "c62ada76533a2de045d4c6062988ba84df140729",
    "previousToHash": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
    "removed": Object {
      "commits": Array [],
      "total": 0,
    },
    "toHash": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "foo@bar.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "COMMENTED",
    "comment": Object {
      "author": Object {
        "active": true,
        "displayName": "DangerCI",
        "emailAddress": "foo@bar.com",
        "id": 2,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/danger",
            },
          ],
        },
        "name": "danger",
        "slug": "danger",
        "type": "NORMAL",
      },
      "comments": Array [],
      "createdDate": 1518939353345,
      "id": 10,
      "permittedOperations": Object {
        "deletable": true,
        "editable": false,
      },
      "properties": Object {
        "repositoryId": 1,
      },
      "tasks": Array [],
      "text": "

| 🚫 | Fails
| --- | --- |

> node\` failed.


| ⚠️ | Warnings
| --- | --- |

> Please add a changelog entry for your changes. You can find it in \`CHANGELOG.md\` 
> 
>  Please add your change and name to the master section.

> 0

> \\"use strict\\";
> 
> module.exports = {
>   runner: \\"jest-runner-eslint\\",
>   displayName: \\"lint\\",
>   testMatch: [\\"<rootDir>/**/*.js\\"],
>   testPathIgnorePatterns: [\\"node_modules/\\"]
> };



### Log


\`\`\`sh
Jest tests passed :+1:
Error:  SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (/Users/azz/code/danger-js/distribution/platforms/git/gitJSONToGitDSL.js:72:60)
    at step (/Users/azz/code/danger-js/distribution/platforms/git/gitJSONToGitDSL.js:32:23)
    at Object.next (/Users/azz/code/danger-js/distribution/platforms/git/gitJSONToGitDSL.js:13:53)
    at fulfilled (/Users/azz/code/danger-js/distribution/platforms/git/gitJSONToGitDSL.js:4:58)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

\`\`\`



|    |
|---:|
| _Generated by 🚫 [dangerJS](http://github.com/danger/danger-js/)_ |


[](http://danger-id-default;)
",
      "updatedDate": 1519449132488,
      "version": 23,
    },
    "commentAction": "ADDED",
    "createdDate": 1518939353345,
    "id": 52,
    "user": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/danger",
          },
        ],
      },
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "UPDATED",
    "addedReviewers": Array [],
    "createdDate": 1518937747368,
    "id": 43,
    "removedReviewers": Array [
      Object {
        "active": true,
        "displayName": "DangerCI",
        "emailAddress": "foo@bar.com",
        "id": 2,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/danger",
            },
          ],
        },
        "name": "danger",
        "slug": "danger",
        "type": "NORMAL",
      },
    ],
    "user": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/danger",
          },
        ],
      },
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "UPDATED",
    "addedReviewers": Array [
      Object {
        "active": true,
        "displayName": "DangerCI",
        "emailAddress": "foo@bar.com",
        "id": 2,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/danger",
            },
          ],
        },
        "name": "danger",
        "slug": "danger",
        "type": "NORMAL",
      },
    ],
    "createdDate": 1518929751675,
    "id": 42,
    "removedReviewers": Array [],
    "user": Object {
      "active": true,
      "displayName": "DangerCI",
      "emailAddress": "foo@bar.com",
      "id": 2,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/danger",
          },
        ],
      },
      "name": "danger",
      "slug": "danger",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "COMMENTED",
    "comment": Object {
      "author": Object {
        "active": true,
        "displayName": "test",
        "emailAddress": "foo@bar.com",
        "id": 1,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/test",
            },
          ],
        },
        "name": "test",
        "slug": "test",
        "type": "NORMAL",
      },
      "comments": Array [],
      "createdDate": 1518870689956,
      "id": 2,
      "permittedOperations": Object {
        "deletable": true,
        "editable": true,
      },
      "properties": Object {
        "repositoryId": 1,
      },
      "tasks": Array [],
      "text": "wow",
      "updatedDate": 1518870689956,
      "version": 0,
    },
    "commentAction": "ADDED",
    "commentAnchor": Object {
      "diffType": "EFFECTIVE",
      "fileType": "TO",
      "fromHash": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
      "line": 1,
      "lineType": "ADDED",
      "orphaned": false,
      "path": "banana",
      "toHash": "7fe85eb2a4dbaa70414155022abc7e5465f09547",
    },
    "createdDate": 1518870689956,
    "diff": Object {
      "destination": Object {
        "components": Array [
          "banana",
        ],
        "name": "banana",
        "parent": "",
        "toString": "banana",
      },
      "hunks": Array [
        Object {
          "destinationLine": 1,
          "destinationSpan": 1,
          "segments": Array [
            Object {
              "lines": Array [
                Object {
                  "commentIds": Array [
                    2,
                  ],
                  "destination": 1,
                  "line": "bar",
                  "source": 0,
                  "truncated": false,
                },
              ],
              "truncated": false,
              "type": "ADDED",
            },
          ],
          "sourceLine": 0,
          "sourceSpan": 0,
          "truncated": false,
        },
      ],
      "properties": Object {
        "current": true,
        "fromHash": "8942a1f75e4c95df836f19ef681d20a87da2ee20",
        "toHash": "7fe85eb2a4dbaa70414155022abc7e5465f09547",
      },
      "source": null,
      "truncated": false,
    },
    "id": 5,
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "foo@bar.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "COMMENTED",
    "comment": Object {
      "author": Object {
        "active": true,
        "displayName": "test",
        "emailAddress": "foo@bar.com",
        "id": 1,
        "links": Object {
          "self": Array [
            Object {
              "href": "http://localhost:7990/users/test",
            },
          ],
        },
        "name": "test",
        "slug": "test",
        "type": "NORMAL",
      },
      "comments": Array [],
      "createdDate": 1518870515570,
      "id": 1,
      "permittedOperations": Object {
        "deletable": true,
        "editable": true,
      },
      "properties": Object {
        "repositoryId": 1,
      },
      "tasks": Array [],
      "text": "heyo",
      "updatedDate": 1518870515570,
      "version": 0,
    },
    "commentAction": "ADDED",
    "createdDate": 1518870515570,
    "id": 4,
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "foo@bar.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
  Object {
    "action": "OPENED",
    "createdDate": 1518863923424,
    "id": 3,
    "user": Object {
      "active": true,
      "displayName": "test",
      "emailAddress": "foo@bar.com",
      "id": 1,
      "links": Object {
        "self": Array [
          Object {
            "href": "http://localhost:7990/users/test",
          },
        ],
      },
      "name": "test",
      "slug": "test",
      "type": "NORMAL",
    },
  },
]
`;
